includes/Core/Email_Reporting/Email_Log.php

```diff
@@ -0,0 +1,383 @@
+		$this->register_post_type();
+		$this->register_post_statuses();
+		$this->register_post_meta();
+	}
+
+	/**
+	 * Registers the internal email log post type.
+	 *
+	 * @since n.e.x.t
+	 */
+	protected function register_post_type() {
+		if ( post_type_exists( self::POST_TYPE ) ) {
+			return;
+		}
+
+		register_post_type(
+			self::POST_TYPE,
+			array(
+				'labels'              => array(
+					'name'               => _x( 'Email Reports', 'Post type general name', 'google-site-kit' ),
+					'singular_name'      => _x( 'Email Report', 'Post type singular name', 'google-site-kit' ),
+					'add_new'            => _x( 'Add New', 'email report', 'google-site-kit' ),
+					'add_new_item'       => __( 'Add New Email Report', 'google-site-kit' ),
+					'edit_item'          => __( 'Edit Email Report', 'google-site-kit' ),
+					'new_item'           => __( 'New Email Report', 'google-site-kit' ),
+					'view_item'          => __( 'View Email Report', 'google-site-kit' ),
+					'search_items'       => __( 'Search Email Reports', 'google-site-kit' ),
+					'not_found'          => __( 'No Email Reports found.', 'google-site-kit' ),
+					'not_found_in_trash' => __( 'No Email Reports found in Trash.', 'google-site-kit' ),
+				),
+				'public'              => false,
+				'show_ui'             => false,
+				'show_in_rest'        => false,
+				'has_archive'         => false,
+				'rewrite'             => false,
+				'publicly_queryable'  => false,
+				'exclude_from_search' => true,
+				'query_var'           => false,
+				'show_in_nav_menus'   => false,
+				'show_in_menu'        => false,
+				'map_meta_cap'        => true,
+				'capability_type'     => 'post',
+				'supports'            => array( 'author' ),
+			)
+		);
+	}
+
+	/**
+	 * Registers internal delivery statuses.
+	 *
+	 * @since n.e.x.t
+	 */
+	protected function register_post_statuses() {
+		$statuses = array(
+			'googlesitekit_email_sent'      => array(
+				'label'       => __( 'Sent', 'google-site-kit' ),
+				/* translators: %s: Number of sent email reports. */
+				'label_count' => _n_noop(
+					'Sent <span class="count">(%s)</span>',
+					'Sent <span class="count">(%s)</span>',
+					'google-site-kit'
+				),
+			),
+			'googlesitekit_email_failed'    => array(
+				'label'       => __( 'Failed', 'google-site-kit' ),
+				/* translators: %s: Number of failed email reports. */
+				'label_count' => _n_noop(
+					'Failed <span class="count">(%s)</span>',
+					'Failed <span class="count">(%s)</span>',
+					'google-site-kit'
+				),
+			),
+			'googlesitekit_email_scheduled' => array(
+				'label'       => __( 'Scheduled', 'google-site-kit' ),
+				/* translators: %s: Number of scheduled email reports. */
+				'label_count' => _n_noop(
+					'Scheduled <span class="count">(%s)</span>',
+					'Scheduled <span class="count">(%s)</span>',
+					'google-site-kit'
+				),
+			),
+		);
+
+		foreach ( $statuses as $slug => $status ) {
+			register_post_status(
```

Comment:
Correct me if I am wrong, but when we register a new post status, it can be applied to all post types that support statuses, right? In this case, I think we should have a pseudo-status approach where statuses are saved as post meta.

Reply:
@zutigrm, please, remove labels for statuses since they are private and labels won't be used.